search_engine_parser/core/base.py

```diff
@@ -11,141 +11,146 @@
+		"""
+		Runs every entry on the page through parse_single_result
+
+		:param results: Result of main search to extract individual results
+		:type results: list[`bs4.element.ResultSet`]
+		:returns: dictionary. Containing titles, links and descriptions.
+		:rtype: dict
+		"""
+		titles = []
+		links = []
+		descs = []
+		for each in results:
+			title = link = desc = " "
+			try:
+				title, link, desc = self.parse_single_result(each)
+				# Append links and text to a list
+				titles.append(title)
+				links.append(link)
+				descs.append(desc)
+			except Exception as e:
+				print(e)
+		search_results = {'titles': titles,
+						'links': links,
+						'descriptions': descs}
+		return search_results
+	
+	@staticmethod
+	def parse_query(query):
+		"""
+		Replace spaces in query
+
+		:param query: query to be processed
+		:type query: str
+		:rtype: str
+		"""
+		return query.replace(" ", "%20")
+	
+	@staticmethod
+	def getSource(url):
+		"""
+		Returns the source code of a webpage.
+
+		:rtype: string
+		:param url: URL to pull it's source code
+		:return: html source code of a given URL.
+		"""
+		# headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:20.0) Gecko/20100101 Firefox/20.0'}
+		# prevent caching
+		user_agent_list = [
+			"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.7; rv:11.0) Gecko/20100101 Firefox/11.0",
+			"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36",
+			"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:22.0) Gecko/20100 101 Firefox/22.0",
+			"Mozilla/5.0 (Windows NT 6.1; rv:11.0) Gecko/20100101 Firefox/11.0",
+			"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_7_4) AppleWebKit/536.5 (KHTML, like Gecko) Chrome/19.0.1084.46 Safari/536.5",
+			"Mozilla/5.0 (Windows; Windows NT 6.1) AppleWebKit/536.5 (KHTML, like Gecko) Chrome/19.0.1084.46 Safari/536.5",
+			]
+		headers = {
+			"Cache-Control": 'no-cache',
+			"Connection": "keep-alive",
+			"User-Agent": random.choice(user_agent_list),
+		}
+		try:
+			response = requests.get(url, headers=headers)
+			html = response.text
+		except Exception as e:
+			raise Exception('ERROR: {}\n'.format(e))
+		return str(html)
+
+	def get_soup(self, url):
+		"""
+		Get the html soup of a query
+
+		:rtype: `bs4.element.ResultSet`
+		"""
+		html = self.getSource(url)
+		return BeautifulSoup(html, 'lxml')
+
+	def get_search_url(self, query=None, page=None):
+		""" 
+		Return a formatted search url
+		"""
+		# Some URLs use offsets
+		offset = (page * 10) - 9
+		
+		return  self.search_url.format(query=query, page=page, offset=offset) 
+
+	def search(self, query=None, page=None):
+		""" 
+		Query the search engine
+
+		:param query: the query to search for 
+		:type query: str
+		:param page: Page to be displayed, defaults to 1
+		:type page: int
+		:return: dictionary. Containing titles, links, netlocs and descriptions.
+		"""
+		parsed_query = self.parse_query(query)
+		
+		if self.name=="Baidu":
```

Comment:
The base class already have a [get_search_url](https://github.com/bisoncorps/search-engine-parser/blob/8c40379bab335e72a9087ba5cf55c08df7a0be1f/search_engine_parser/core/base.py#L121) function. This function receives the the query and page as argument. You should rather override this function to return the correct url, instead of reconstructing the `search` function.

Reply:
Noted, will do so.